zerver/webhooks/gitlab/view.py

```diff
@@ -382,12 +387,16 @@ def get_subject_based_on_event(event: str, payload: Dict[str, Any]) -> str:
 
 def get_event(request: HttpRequest, payload: Dict[str, Any], branches: Optional[str]) -> Optional[str]:
     event = validate_extract_webhook_http_header(request, 'X_GITLAB_EVENT', 'GitLab')
+    if event == "System Hook":
+        event_name = payload.get('event_name', payload.get('object_kind'))
+        event = fixture_to_headers(event_name).get("HTTP_X_GITLAB_EVENT", event)
```

Comment:
I don't understand why you need to call fixture_to_headers here; can you explain?

Reply:
fixture_to_headers goes from an event_name like `merge_request` which we get in the payload, to `Merge Request`, which is what we normally see in the `X-Gitlab-Event` header from project hooks. This means it can leverage the existing logic in the function.

I could perform the logic at this point, but saw there was a function that does the same thing. Would it be better to copy the logic here?

Reply:
Simplified to 
```python
    if event == "System Hook":
        # Convert the event name to a Gitlab event title
        event_name = payload.get('event_name', payload.get('object_kind'))
        event = event_name.split("__")[0].replace("_", " ").title()
        event = f"{event} Hook"
```

Reply:
I'm a little suspicious that this logic is a testing artifact, resulting from your naming the files improperly.  The naming convention should be that the part before the `__` in the filename encodes the HTTP header for the GitLab event type ... and the rest of the filename you can choose freely.

I assume GitLab is sending as `X-GitLab-Event` just "System Hook", right?  So I think these fixtures should all be named `system_hook__rest_of_filename`.

Reply:
You're right. I've fixed the files and test cases to work correctly, and also removed the conditional change from `test_classes.py` in the commit.